src/mantarray_desktop_app/mc_simulator.py

```diff
@@ -111,6 +86,15 @@ def __init__(
         self._status_code_bits: bytes = bytes(0)
         self._reset_status_code_bits()
 
+    @property
+    def in_waiting(self) -> int:
+        if self._leftover_read_bytes is None:
+            try:
+                self._leftover_read_bytes = self._output_queue.get_nowait()
```

Comment:
is there any chance of this behaving in unexpected ways in unit tests if the `read_timeout_seconds` is set to a non-zero value?